api/secrets.go

```diff
@@ -0,0 +1,133 @@
+	savedSecret, err := model.Kube.ApplySecret(model.Env.RadixDeploymentNamespace, secret)
+	return savedSecret, err
+}
+
+//CreateBatchScheduleDescriptionSecret Create a secret for the batch schedule description
+func (model *Model) CreateBatchScheduleDescriptionSecret(batchName string, jobComponent *v1.RadixDeployJobComponent, rd *v1.RadixDeployment, batchScheduleDescription *models.BatchScheduleDescription) (*corev1.Secret, error) {
+	secret, err := buildBatchScheduleDescriptionSecretSpec(batchName, rd.Spec.AppName, jobComponent.Name, batchScheduleDescription)
+	if err != nil {
+		return nil, err
+	}
+	return model.Kube.ApplySecret(model.Env.RadixDeploymentNamespace, secret)
+}
+
+//GetSecretsForJob Get secrets for the job
+func (model *Model) GetSecretsForJob(jobName string) (*corev1.SecretList, error) {
+	return model.KubeClient.CoreV1().Secrets(model.Env.RadixDeploymentNamespace).List(
+		context.TODO(),
+		metav1.ListOptions{
+			LabelSelector: getLabelSelectorForSecret(jobName, model.Env.RadixComponentName),
+		},
+	)
+}
+
+//DeleteSecret Delete the service for the job
+func (model *Model) DeleteSecret(secret *corev1.Secret) error {
+	return model.KubeClient.CoreV1().Secrets(secret.Namespace).Delete(context.TODO(), secret.Name, metav1.DeleteOptions{})
+}
+
+func getLabelSelectorForSecret(jobName, componentName string) string {
+	return labels.SelectorFromSet(map[string]string{
+		kube.RadixComponentLabel: componentName,
+		kube.RadixJobNameLabel:   jobName,
+	}).String()
+}
+
+func buildPayloadSecretSpec(secretName, payload, jobName, appName, componentName string) *corev1.Secret {
+	secret := corev1.Secret{
+		Type: corev1.SecretTypeOpaque,
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secretName,
+			Labels: map[string]string{
+				kube.RadixAppLabel:       appName,
+				kube.RadixComponentLabel: componentName,
+				kube.RadixJobTypeLabel:   kube.RadixJobTypeJobSchedule,
+				kube.RadixJobNameLabel:   jobName,
+			},
+		},
+		Data: map[string][]byte{
+			defaults.JobPayloadPropertyName: []byte(payload),
+		},
+	}
+	return &secret
+}
+
+func isPayloadDefinedForJobComponent(radixJobComponent *v1.RadixDeployJobComponent) bool {
+	return radixJobComponent.Payload != nil && strings.TrimSpace(radixJobComponent.Payload.Path) != ""
+}
+
+func buildBatchScheduleDescriptionSecretSpec(batchName, appName, componentName string, batchScheduleDescription *models.BatchScheduleDescription) (*corev1.Secret, error) {
+	secretName := defaults.GetBatchScheduleDescriptionSecretName(batchName)
+	descriptionJson, err := json.Marshal(batchScheduleDescription)
+	if err != nil {
+		return nil, err
+	}
+
+	secret := corev1.Secret{
+		Type: corev1.SecretTypeOpaque,
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secretName,
+			Labels: map[string]string{
+				kube.RadixAppLabel:       appName,
+				kube.RadixComponentLabel: componentName,
+				kube.RadixJobTypeLabel:   kube.RadixJobTypeJobSchedule,
```

Comment:
It kube.RadixJobTypeJobSchedule correct, or should it be kube.RadixJobTypeBatchSchedule?